### DISCUSSION TITLE
Is it possible to run mocks in contract test?

### DISCUSSION THREAD

**Discussion by GregLeung** [specmatic/specmatic#814] [Ask For Help]
In Java projects, Is it possible to run mocks in contract test?

**Comment by joelrosario** [comment]
Hi @GregLeung, [here's an example](https://github.com/znsio/specmatic-order-bff-java/blob/1b1c309a4c7866ff5d1e4df66a2e764f027290c7/src/test/kotlin/com/component/orders/api/APITests.kt#L36C10-L36C10) of how this could be done. The `createStub` function is also [documented here](https://specmatic.in/documentation/service_virtualization_tutorial.html#programmatically-starting-stub-server-within-tests).

You'd need to add the specmatic-core dependency to your `build.gradle` or `pom.xml` file.

Does this help?

**Comment by GregLeung** [comment]
Hi, thanks for the info, May I ask if Mockito in particular is supported or not? cheers
